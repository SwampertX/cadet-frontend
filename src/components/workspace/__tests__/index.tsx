import * as React from 'react'

import { shallow } from 'enzyme'

import Workspace, { WorkspaceProps } from '..'
import { SideContentTab } from '../side-content'

const playgroundIntroductionTab: SideContentTab = {
  label: 'Introduction',
  icon: '',
  body: <p />
}

const listVisualizerTab: SideContentTab = {
  label: 'List Visualizer',
  icon: '',
  body: <p />
}

test('Workspace renders correctly', () => {
  const props: WorkspaceProps = {
    controlBarProps: {
      externalLibraryName: '',
      handleChapterSelect: () => {},
      handleExternalSelect: () => {},
      handleEditorEval: () => {},
      handleGenerateLz: () => {},
      handleInterruptEval: () => {},
      handleReplEval: () => {},
      handleReplOutputClear: () => {},
      hasChapterSelect: true,
      hasSaveButton: false,
      hasShareButton: true,
      isRunning: true,
      queryString: '/playground',
      questionProgress: null,
      sourceChapter: 3
    },
    editorProps: {
      editorValue: '',
      handleEditorEval: () => {},
      handleEditorValueChange: () => {}
    },
    editorWidth: '100%',
    handleEditorWidthChange: () => {},
    handleSideContentHeightChange: () => {},
    replProps: {
      output: [],
      replValue: '',
      handleBrowseHistoryDown: () => {},
      handleBrowseHistoryUp: () => {},
      handleReplEval: () => {},
      handleReplValueChange: () => {}
    },
    sideContentHeight: 10,
    sideContentProps: {
      activeTab: 0,
      handleChangeActiveTab: () => {},
      tabs: [playgroundIntroductionTab, listVisualizerTab]
    }
  }

  const app = <Workspace {...props} />
  const tree = shallow(app)
  expect(tree.debug()).toMatchSnapshot()
})
