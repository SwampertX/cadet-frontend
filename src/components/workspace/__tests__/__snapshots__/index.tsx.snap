// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workspace renders correctly 1`] = `
"<div className=\\"workspace\\">
  <ControlBar externalLibraryName=\\"\\" handleChapterSelect={[Function: handleChapterSelect]} handleExternalSelect={[Function: handleExternalSelect]} handleEditorEval={[Function: handleEditorEval]} handleGenerateLz={[Function: handleGenerateLz]} handleInterruptEval={[Function: handleInterruptEval]} handleReplEval={[Function: handleReplEval]} handleReplOutputClear={[Function: handleReplOutputClear]} hasChapterSelect={true} hasSaveButton={false} hasShareButton={true} isRunning={true} queryString=\\"/playground\\" questionProgress={{...}} sourceChapter={3} hasUnsavedChanges={[undefined]} onClickNext={[Function: onClickNext]} onClickPrevious={[Function: onClickPrevious]} onClickSave={[Function: onClickSave]} />
  <div className=\\"row workspace-parent\\">
    <div className=\\"editor-divider\\" />
    <Resizable className=\\"resize-editor left-parent\\" enable={{...}} minWidth={0} onResize={[Function]} onResizeStop={[Function: onResizeStop]} size={{...}} onResizeStart={[Function: onResizeStart]} style={{...}} grid={{...}} lockAspectRatio={false} lockAspectRatioExtraWidth={0} lockAspectRatioExtraHeight={0}>
      <Editor editorValue=\\"\\" handleEditorEval={[Function: handleEditorEval]} handleEditorValueChange={[Function: handleEditorValueChange]} />
    </Resizable>
    <div className=\\"right-parent\\">
      <Resizable bounds=\\"parent\\" className=\\"resize-side-content\\" enable={{...}} minHeight={0} onResize={[Function]} onResizeStop={[Function: onResizeStop]} size={{...}} onResizeStart={[Function: onResizeStart]} style={{...}} grid={{...}} lockAspectRatio={false} lockAspectRatioExtraWidth={0} lockAspectRatioExtraHeight={0}>
        <SideContent activeTab={0} handleChangeActiveTab={[Function: handleChangeActiveTab]} tabs={{...}} />
        <div className=\\"side-content-divider\\" />
      </Resizable>
      <Repl output={{...}} replValue=\\"\\" handleBrowseHistoryDown={[Function: handleBrowseHistoryDown]} handleBrowseHistoryUp={[Function: handleBrowseHistoryUp]} handleReplEval={[Function: handleReplEval]} handleReplValueChange={[Function: handleReplValueChange]} />
    </div>
  </div>
</div>"
`;
